runtime/lua/vim/filetype/detect.lua

```diff
@@ -1172,7 +1172,8 @@ function M.sh(path, contents, name)
       vim.b[b].is_kornshell = nil
       vim.b[b].is_sh = nil
     end
-  elseif matchregex(name, [[\<sh\>]]) then
+    -- Ubuntu links sh to dash
+  elseif matchregex(name, [[\<sh\>]]) or matchregex(name, [[\<dash\>]]) then
```

Comment:
Maybe we can use `elseif matchregex(name, [[\<\(sh|dash\)\>]])` here

Reply:
Probably want to do this a bit before for `bash` as well then?